### ISSUE TITLE
False-negative on unused private variables

### ISSUE THREAD

**Issue by KrzysztofKowalskiKainos** [panuhorsmalahti/gulp-tslint#30]
When private variable is accessed only through self variable it produces 'unused variable' error.

Here is short example:
```javascript
class SomeClass {
    constructor() {
         var self: SomeClass = this;
         self.doSomething();
         var newX: number = self.x;
    }

    private x: number;
    private doSomething(): void { 
    ...
    }
}
```

It occurs since 3.1.2 version.

**Comment by panuhorsmalahti** [comment]
This is likely a tslint issue, not gulp-tslint.

**Comment by panuhorsmalahti** [comment]
Can you try this with only tslint without using gulp-tslint?

**Comment by panuhorsmalahti** [comment]
Closing as a tslint issue.
